spec/policy/custom_metrics/custom_metrics_spec.lua

```diff
@@ -0,0 +1,79 @@
+local custom_metrics =  require("apicast.policy.custom_metrics")
+local ngx_variable = require 'apicast.policy.ngx_variable'
+local backend_client = require('apicast.backend_client')
+local Usage = require('apicast.usage')
+
+
+describe('custom metrics policy', function()
+  local context = {}
+  local usage = {}
+
+  before_each(function()
+    ngx.var = {}
+    ngx.header = {}
+
+    stub(ngx_variable, 'available_context', function(context) return context end)
+    stub(ngx.req, 'get_headers', function() return {} end)
+    stub(ngx.resp, 'get_headers', function() return {} end)
+
+    context = { usage = {} }
+    stub(context.usage, 'merge', function() return end)
+
+    usage = Usage.new()
+    usage:add("foo", 1)
+  end)
+
+  it('if Auth cache is disabled, report the usage', function()
```

Comment:
Not too important, but in cases like this, I like to put the condition outside the `it` clause, so the `it` clause reads like natural English. Something like this:
```lua
describe('when the auth cache is disabled', function()
  it('reports the usage', function()
     ...
  )
)
```